code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -267,7 +267,40 @@
 	else
 		M.visible_message("<span class='notice'>[M] hugs [src] to make [p_them()] feel better!</span>", \
 					"<span class='notice'>You hug [src] to make [p_them()] feel better!</span>")
-		SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "hug", /datum/mood_event/hug)
+
+		// Warm them up with hugs
+		if(bodytemperature > M.bodytemperature) // they are warmer leech from them

```

Comment:
This feels like copypasta from the temperature PR. If I'm right, can we make them share the code somehow please?

Reply:
This is not a direct copy, but sharing body heat might be good to get as a proc somewhere in the mob framework. I have an idea let me see.

Reply:
Ok Cyber I did some major changes.
natural_bodytemperature_stabilization is now self contained.
adjust_bodytemperature is expanded for carbon mobs to allow insulation and change steps
handle_environment has been reduced in size as most of it is now done in the other two steps.